lib/internal/process/esm_loader.js

```diff
@@ -60,17 +60,24 @@ async function initializeLoader() {
     cwd = 'file:///';
   }
 
+  const keyedExportsList = [];
+
   // A separate loader instance is necessary to avoid cross-contamination
   // between internal Node.js and userland. For example, a module with internal
   // state (such as a counter) should be independent.
   const internalEsmLoader = new ESMLoader();
 
-  // Importation must be handled by internal loader to avoid poluting userland
-  const keyedExportsList = await internalEsmLoader.import(
-    customLoaders,
-    pathToFileURL(cwd).href,
-    ObjectCreate(null),
-  );
+  for (const customLoader of customLoaders) {
+    // Importation must be handled by internal loader to avoid poluting userland
+    const keyedExportsSublist = await internalEsmLoader.import(
+      [customLoader],
+      pathToFileURL(cwd).href,
+      ObjectCreate(null),
+    );
+
+    await internalEsmLoader.addCustomLoaders(keyedExportsSublist);
+    keyedExportsList.push(...keyedExportsSublist);
```

Comment:
To avoid unsafe array iteration
```suggestion
    ArrayPrototypePushApply(keyedExportsList, keyedExportsSublist);
```